apps/editing-toolkit/editing-toolkit-plugin/global-styles/class-global-styles-fonts-message-control.php

```diff
@@ -0,0 +1,136 @@
+			</p>
+		<?php
+	}
+
+	/**
+	 * Render a link to the global styles fonts page help support document
+	 */
+	private function render_learn_more_link() {
+		// Translators: This is a link which opens: https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles.
+		$learn_more_link_text = __( 'Learn more about changing fonts using Global Styles.', 'full-site-editing' );
+		?>
+			<p>
+				<a
+					href="https://wordpress.com/support/custom-fonts/#changing-fonts-with-global-styles"
+					target="_blank"
+					onClick="<?php echo esc_attr( $this->get_tracks_event_script( 'calypso_customizer_global_styles_support_link_clicked' ) ); ?>"
+				>
+					<?php echo esc_html( $learn_more_link_text ); ?>
+				</a>
+			</p>
+		<?php
+	}
+
+	/**
+	 * Render a link to the global styles section of the block editor, if a homepage exists
+	 */
+	private function maybe_render_block_editor_link() {
+		$block_editor_with_global_styles_url = $this->get_link_to_editor_with_global_styles_sidebar();
+		if ( null === $block_editor_with_global_styles_url ) {
+			return;
+		}
+		// Translators: This is a link which opens the block editor, and then opens the global styles sidebar.
+		$block_editor_link_text = __( 'Click here to open the Block Editor and change your fonts.', 'full-site-editing' );
+		?>
+			<p>
+				<a
+					href="<?php echo esc_url( $block_editor_with_global_styles_url ); ?>"
+					target="_blank"
+					onClick="<?php echo esc_attr( $this->get_tracks_event_script( 'calypso_customizer_global_styles_block_editor_link_clicked' ) ); ?>">
+					<?php echo esc_html( $block_editor_link_text ); ?>
+				</a>
+			</p>
+		<?php
+	}
+
+	/**
+	 * Creates a link to the page editor for the user's homepage with the global styles sidebar opened.
+	 */
+	private function get_link_to_editor_with_global_styles_sidebar() {
+		$homepage_id = \get_option( 'page_on_front' );
+		if ( empty( $homepage_id ) ) {
+			return null;
+		}
+
+		$base_url = null;
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( 'http://calypso.localhost:3000' === $_GET['calypsoOrigin'] ) {
+			$base_url = 'http://calypso.localhost:3000/';
+		} else {
+			$base_url = 'https://www.wordpress.com/';
+		}
+
+		$site_slug = $this->get_site_slug();
+
+		$url_components = array(
+			$base_url,
+			'page/',
+			$site_slug . '/',
+			$homepage_id,
+			'?openSidebar=global-styles',
+		);
+
+		return implode( '', $url_components );
+	}
+
+	/**
+	 * Get the site slug
+	 */
+	private function get_site_slug() {
+		if ( method_exists( '\WPCOM_Masterbar', 'get_calypso_site_slug' ) ) {
+			return \WPCOM_Masterbar::get_calypso_site_slug( get_current_blog_id() );
+		}
+
+		$home_url  = home_url( '/' );
+		$site_slug = wp_parse_url( $home_url, PHP_URL_HOST );
+
+		return $site_slug;
+	}
+
+	/**
+	 * Returns javascript which creates a tracks event
+	 *
+	 * @param string $tracks_event The name of the tracks event.
+	 */
+	private function get_tracks_event_script( $tracks_event ) {
```

Comment:
I admit using concatenation for piecing together a script worries me, as it can be very fragile. It would be better to enqueue a standalone script, which is more "the WordPress way", and expected by other WP devs. https://developer.wordpress.org/reference/functions/wp_enqueue_script/

Reply:
I agree, it's a mess. I was rushing when my previous implementation didn't test well on atomic right before the sprint demo. The newer version should be better, I hope.